### DISCUSSION TITLE
Make a table blank

### DISCUSSION THREAD

**Discussion by Andybonner300** [filamentphp/filament#12386] [Help]
### Package

Table builder

### Package Version

3.2

### How can we help you?

Is it possible for a table to be blank until a search has been made,. 

in this table 
public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('Our_Ref') 
                ->searchable(),
                Tables\Columns\TextColumn::make('Company_Name')
                ->searchable(),
                Tables\Columns\TextColumn::make('Advert_Code'), 
                Tables\Columns\TextColumn::make('Cost')
                ->money('gbp'), 
                Tables\Columns\TextColumn::make('Feature_Name'), 
                Tables\Columns\TextColumn::make('Rep_No'), 
            ])

I don't want any of the information to show until you a have searched Company_Name.

**Comment by leandrocfe** [comment]
1 -You could create a SelectFilter to show the companies with `->searchable()`
https://filamentphp.com/docs/3.x/tables/filters/select

2 - use `modifyQueryUsing` to customize the query
```php
$table
    ->modifyQueryUsing(function (Builder $query, Table $table) {
        $query->when(! $table->isFiltered(), fn (Builder $query): Builder => $query->whereId(0));
    })
```

**Reply by flightphysical** [reply]
I like this idea...where would you place the modification to prevent the table from reseting to blank state after each real query/filter?
